pkg/compact/compact.go

```diff
@@ -1020,6 +1029,39 @@ func (cg *Group) areBlocksOverlapping(include *metadata.Meta, exclude ...*metada
 	return nil
 }
 
+func (cg *Group) removeOverlappingBlocks(ctx context.Context, toCompact []*metadata.Meta, dir string, blockDeletableChecker BlockDeletableChecker) error {
+	if len(toCompact) == 0 {
+		return nil
+	}
+	kept := toCompact[0]
+	for _, m := range toCompact {
+		if m.MinTime < kept.MinTime && m.MaxTime > kept.MaxTime {
+			level.Warn(cg.logger).Log("msg", "found overlapping block in plan that are not the first",
+				"first", kept.String(), "block", m.String())
+			kept = m
+		} else if m.MinTime < kept.MinTime || m.MaxTime > kept.MaxTime {
```

Comment:
Two disjoint blocks (e.g., [0, 1) and [2, 3)) will trigger this half as well. Should it be
```
if kept contains m {
} else if m contains kept {
} else {
  halt;
}

Reply:
It comes from this group:
01HQF7C1GKTJZSW7B0BWTRS6K0 (min time: 1708732800000, max time: 1708761600000) 01HQD12MFPJ9EYMQGXRFWFHNE3 (min time: 1708740000000, max time: 1708761600000) 01HQHWVPCPH519JQN3RC2HRM6D (min time: 1708747200000, max time: 1708819200000)

Reply:
I meant the condition ` if m.MinTime < kept.MinTime || m.MaxTime > kept.MaxTime ` includes the disjoint case besides overlapping case.